### PR TITLE
allow to select the archive to restore from

### LINKED ISSUE #39: Allow to select the borg archive to restore from
Currently it is only possible to restore the most recent backup. Make it possible to select an archive.

### PR DESCRIPTION
**Pull request by szaimen**
Close #39

Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

**Comment by szaimen** [comment]
This is now finally ready for review :)

File: php/src/Data/ConfigurationManager.php [hunk 1]
```diff
@@ -83,6 +83,30 @@ public function GetLastBackupTime() : string {
         return $lastBackupTime;
     }
 
+    public function GetBackupTimes() : array {
+        if (!file_exists(DataConst::GetBackupArchivesList())) {
+            return $array[] = '';
+        }
+        
+        $content = file_get_contents(DataConst::GetBackupArchivesList());
+        if ($content === "") {
```
Comment (CarlSchwan) [inline php/src/Data/ConfigurationManager.php:92]:
```suggestion
        if (count($content) === 0) {
```

File: php/src/Data/ConfigurationManager.php [hunk 2]
```diff
@@ -83,6 +83,30 @@ public function GetLastBackupTime() : string {
...
+            return $array[] = '';
+        }
+
+        $backupLines = explode("\n", $content);
```
Comment (CarlSchwan) [inline php/src/Data/ConfigurationManager.php:96]:
```suggestion
        $backupLines = explode('\n', $content);
```

To be consistent with the rest of the code

File: php/src/Data/ConfigurationManager.php [hunk 3]
```diff
@@ -83,6 +83,30 @@ public function GetLastBackupTime() : string {
...
+        $backupTimes = array();
```
Comment (CarlSchwan) [inline php/src/Data/ConfigurationManager.php:97]:
```suggestion
        $backupTimes = [];
```

File: php/src/Data/ConfigurationManager.php [hunk 4]
```diff
@@ -83,6 +83,30 @@ public function GetLastBackupTime() : string {
...
+        foreach($backupLines as $lines) {
+            $backupTimesTemp = explode(",", $lines);
```
Comment (CarlSchwan) [inline php/src/Data/ConfigurationManager.php:99]:
```suggestion
            $backupTimesTemp = explode(',', $lines);
```

**Review by CarlSchwan — COMMENTED** [review]
Some comments on the PHP code

**Comment by szaimen** [comment]
Thanks for the review Carl! :)

**Pull request merged**